src/app/event/Modal.tsx

```diff
@@ -0,0 +1,117 @@
+import { useState } from "react";
+import { isAddress } from "viem";
+import {
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+} from "@/components/ui/dialog";
+import { useFieldArray, useForm } from "react-hook-form";
+
+import {
+  AssignTicketText,
+  CloseDialogButton,
+  TriggerDialogButton,
+} from "./components/ModalComponents";
+
+interface FormData {
+  addresses: { address: string }[];
+}
+
+export function Modal() {
+  const [open, setOpen] = useState(false);
+  const {
+    register,
+    control,
+    handleSubmit,
+    watch,
+    formState: { errors, isValid },
+  } = useForm<FormData>({
+    defaultValues: {
+      addresses: [{ address: "" }],
+    },
+    mode: "onChange",
+  });
+
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: "addresses",
+  });
+
+  const watchedFields = watch("addresses");
+
+  function submitForm(data: FormData) {
+    const filteredData = data.addresses.filter(
+      (element) => element.address !== ""
+    );
+    console.log("Form submitted with data:", filteredData);
+    setOpen(false);
+  }
+
+  function handleNumberOfFields() {
+    const lastElement = watchedFields[watchedFields.length - 1];
+    if (isAddress(lastElement?.address)) {
+      append({ address: "" });
+    }
+
+    const emptyElements = watchedFields
+      .map((field, index) => (field.address === "" ? index : -1))
+      .filter((index) => index !== -1);
+    if (emptyElements.length > 1) {
+      remove(emptyElements[0]);
+    }
+  }
+
+  return (
+    <Dialog open={open} onOpenChange={setOpen}>
+      <DialogTrigger asChild>
+        <TriggerDialogButton />
+      </DialogTrigger>
+      <DialogContent className="max-w-fit">
+        <DialogHeader>
+          <DialogTitle>
+            <AssignTicketText />
+          </DialogTitle>
+          <DialogDescription className="m-0">
+            <br />
+            <strong className="text-black">Attendee</strong>
+            <br />
+            <br />
+            <p className="text-slate-800 text-base">
+              Paste attendee{"'"}s wallet address
+            </p>
+            <br />
+            <form onSubmit={handleSubmit(submitForm)}>
+              {fields.map((field, index) => (
+                <div key={field.id}>
+                  <input
```

Comment:
![image](https://github.com/user-attachments/assets/70d9748f-9e68-4ba7-8a54-7c8f67e60a15)
I would suggest to align the inputs to the center, imo it's a bit weird this empty space on the righ side and the submit button center aligned.